collect_app/src/androidTest/java/org/odk/collect/android/benchmark/EntitiesBenchmarkTest.kt

```diff
@@ -0,0 +1,120 @@
+package org.odk.collect.android.benchmark
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.blankOrNullString
+import org.hamcrest.Matchers.lessThan
+import org.hamcrest.Matchers.not
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import org.odk.collect.android.support.TestDependencies
+import org.odk.collect.android.support.pages.MainMenuPage
+import org.odk.collect.android.support.pages.Page
+import org.odk.collect.android.support.rules.CollectTestRule
+import org.odk.collect.android.support.rules.TestRuleChain.chain
+
+private const val PROJECT_URL = ""
+
+/**
+ * Benchmarks the performance of entity follow up forms. [PROJECT_URL] should be set to a project
+ * that contains the "100k Entities Filter" form.
+ *
+ * Devices that currently pass:
+ * - Pixel 4a
+ * - Fairphone 3
+ *
+ */
+
+@RunWith(AndroidJUnit4::class)
+class EntitiesBenchmarkTest {
+
+    private val rule = CollectTestRule(useDemoProject = false)
+
+    @get:Rule
+    var chain: RuleChain = chain(TestDependencies(true)).around(rule)
+
+    @Test
+    fun run() {
+        assertThat("Need to set PROJECT_URL before running!", PROJECT_URL, not(blankOrNullString()))
+        clearAndroidCache() // Make sure HTTP cache doesn't affect anything
```

Comment:
It actually probably makes sense to delete the cache, download the form, delete it and then download it again to make sure that we can calculate the download time without the network cost.